src/core/tsi/ssl_transport_security.cc

```diff
@@ -1955,16 +1954,15 @@ tsi_result tsi_create_ssl_server_handshaker_factory_with_options(
 
   for (i = 0; i < options->num_key_cert_pairs; i++) {
     do {
-#if defined(OPENSSL_NO_TLS1_2_METHOD) || OPENSSL_API_COMPAT >= 0x10100000L
       impl->ssl_contexts[i] = SSL_CTX_new(TLS_method());
-#else
-      impl->ssl_contexts[i] = SSL_CTX_new(TLSv1_2_method());
-#endif
       if (impl->ssl_contexts[i] == nullptr) {
         gpr_log(GPR_ERROR, "Could not create ssl context.");
         result = TSI_OUT_OF_RESOURCES;
         break;
       }
+#if OPENSSL_VERSION_NUMBER >= 0x10100000
+      SSL_CTX_set_min_proto_version(impl->ssl_contexts[i], TLS1_2_VERSION);
```

Comment:
What's the default minimum supported version if we do not set it explicitly here?

Reply:
If we don't set, then it is really up to the SSL library, most likely version TLS 1.1 or below.

Reply:
Thanks for the confirmation.

Reply:
If we set it like this, do not have a backward-compatibility issue when a client or server with TLS 1.1 or below wants to talk to its peer whose minimum supported version is TLS 1.2 or above?

Reply:
Thanks for taking care of it.

Reply:
In previous code, if user does not config any thing, it will use TLSv1_2_method(). 
The only way to use TLS 1.1 before was set OPENSSL_NO_TLS1_2_METHOD and use an SSL library thats does not support TLS 1.2 or above. 

The question is whether we want to provide better security (to prevent downgrade attack) or be full backward compatible.

Let me remove this and add a todo.